unstructured/ingest/connector/sharepoint.py

```diff
@@ -168,7 +184,44 @@ def _fetch_page(self):
             return None
         return page
 
+    # todo: improve permissions - ingestdoc matching logic
+    def update_permissions_data(self):
+        def parent_name_matches(parent_type, permissions_filename, ingest_doc_filepath):
+            permissions_filename = permissions_filename.split("_SEP_")
+            ingest_doc_filepath = ingest_doc_filepath.split("/")
+
+            if parent_type == "sites":
+                return permissions_filename[0] == ingest_doc_filepath[1]
+
+            elif parent_type == "SitePages" or parent_type == "Shared Documents":
+                return True
+
+        self._permissions_data = None
+        permissions_dir = Path(self.partition_config.output_dir) / "permissions_data"
+
+        if permissions_dir.is_dir():
+            print("FILE_PATH", self.file_path)
+            parent_type = self.file_path.split("/")[0]
+
+            if parent_type == "sites":
+                read_dir = permissions_dir / "sites"
+            elif parent_type == "SitePages" or parent_type == "Shared Documents":
+                read_dir = permissions_dir / "other"
+
+            for filename in os.listdir(read_dir):
+                permissions_docname = os.path.splitext(filename)[0].split("_SEP_")[1]
+                ingestdoc_docname = self.file_path.split("/")[-1]
+
+                if ingestdoc_docname == permissions_docname and parent_name_matches(
+                    parent_type=parent_type,
+                    permissions_filename=filename,
+                    ingest_doc_filepath=self.file_path,
+                ):
+                    with open(read_dir / filename) as f:
+                        self._permissions_data = json.loads(f.read())
+
     def update_source_metadata(self, **kwargs):
+        self.update_permissions_data()
```

Comment:
should updating and setting permissions data be conditional on whether or not the user specified / opted in to it in the command line args?